homeassistant/components/ecovacs/controller.py

```diff
@@ -124,3 +124,8 @@ def devices(self, capability: type[Capabilities]) -> Generator[Device, None, Non
         for device in self._devices:
             if isinstance(device.capabilities, capability):
                 yield device
+
+    @property
+    def legacy_devices(self) -> list[VacBot]:
+        """Return legacy devices."""
+        return self._legacy_devices
```

Comment:
Why this change?

Reply:
Because the mock was not setting it automatically if I used `self.legacy_devices= []` as before.

The tests did not fail but there were an error log that the `legacy_devices` attribute did not exist. I fix it by creating a property, which will be detected by the mock